src/metrics.rs

```diff
@@ -178,45 +232,75 @@ impl Metrics {
         self.window_config.window_start + (self.window_config.window_duration * window_epoch)
     }
 
-    fn add_missing_windows(&mut self, window_epoch: u32) {
+    // returns true if at least one new epoch was added
+    fn add_missing_windows(&mut self, window_epoch: u32) -> bool {
         // Check if the window has already been added.
         if !self.samples.is_empty() && self.samples.back().unwrap().window_epoch >= window_epoch {
-            return;
+            return false;
         }
 
         let start_epoch = self.samples.back().map(|w| w.window_epoch + 1).unwrap_or(0);
         let end_epoch = window_epoch + 1;
 
         for wi in start_epoch..end_epoch {
             self.samples.push(AggregatedSample {
-                hist: Histogram::new(7, 64).unwrap(),
+                msg_rate: Histogram::new(7, 64).unwrap(),
+                report_count: 0,
+                total_success: 0,
+                total_error: 0,
                 window_epoch: wi,
             });
         }
+        start_epoch < end_epoch
     }
 
     fn recv_metrics(&mut self, metrics: AggregatedMetricsMessages) {
         match metrics {
             AggregatedMetricsMessages::AggWindow {
-                messages_in_window,
+                window_counts,
                 window_epoch,
             } => {
-                self.add_missing_windows(window_epoch);
-                // Used to convert messages in window to messages/sec
-                let normalization_c = self.window_config.window_duration.as_secs();
+                let added_epoch = self.add_missing_windows(window_epoch);
+
+                self.total_counts.success_count += window_counts.success_count;
+                self.total_counts.error_count += window_counts.error_count;
+
+                // we use the addition of a new epoch as a convenient "clock" to emit periodic log
+                // messages about the overall progress
+                if added_epoch {
+                    info!(
+                        "Running totals: success={}, errors={}, clients={}/{} (active/total)",
```

Comment:
Logging so we can tell what's going on in a global way (existing logging was per-producer and very limited).